aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -19,6 +19,298 @@ The model classes you create are known as POCO classes (from **P**lain **O**ld *
+```
+
+The `ef migrations add InitialCreate` command generates an *Migrations/{timestamp}_InitialCreate.cs* migration file. Because this is the first migration, the generated class contains code to create the database schema.
+The `ef database update` command updates the database to the last migration. The last migration is the `InitialCreate`  class created with the previous command.
+
+[!INCLUDE [explain warning](~/includes/mvc-intro/model-mig.md)]
+
+[!INCLUDE [ more information on the CLI tools for EF Core](~/includes/ef-cli.md)]
+
+---
+
+### The InitialCreate class
+
+Examine the *Migrations/{timestamp}_InitialCreate.cs* migration file. The `Up` method creates the Movie table and configures `Id` as the primary key. The `Down` method removes the `Up` migration schema changes.
+
+## Examine the context registered with dependency injection
+
+ASP.NET Core is built with [dependency injection (DI)](xref:fundamentals/dependency-injection). Services (such as the EF Core DB context) are registered with DI during application startup. Components that require these services (such as Razor Pages) are provided these services via constructor parameters. The constructor code that gets a DB context instance is shown later in the tutorial.
+
+# [Visual Studio](#tab/visual-studio)
+
+The scaffolding tool automatically created a DB context and registered it with the DI container.
+
+Examine the following `Startup.ConfigureServices` method. The highlighted line was added by the scaffolder:
+
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Startup.cs?name=snippet_ConfigureServices&highlight=6-7)]
+
+The `MvcMovieContext` coordinates EF Core functionality (Create, Read, Update, Delete, etc.) for the `Movie` model. The data context (`MvcMovieContext`) is derived from [Microsoft.EntityFrameworkCore.DbContext](/dotnet/api/microsoft.entityframeworkcore.dbcontext). The data context specifies which entities are included in the data model:
+
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Data/MvcMovieContext.cs)]
+
+The preceding code creates a [DbSet\<Movie>](/dotnet/api/microsoft.entityframeworkcore.dbset-1) property for the entity set. In Entity Framework terminology, an entity set typically corresponds to a database table. An entity corresponds to a row in the table.
+
+The name of the connection string is passed in to the context by calling a method on a [DbContextOptions](/dotnet/api/microsoft.entityframeworkcore.dbcontextoptions) object. For local development, the [ASP.NET Core configuration system](xref:fundamentals/configuration/index) reads the connection string from the *appsettings.json* file.
+
+# [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
+
+You created a DB context and registered it with the DI container.
+
+---
+
+<a name="test"></a>
+
+### Test the app
+
+* Run the app and append `/Movies` to the URL in the browser (`http://localhost:port/movies`).
```

Comment:
earlier it said to just click the Movie App link, and that would be easier here.